lib/ocpp/v201/charge_point.cpp

```diff
@@ -679,19 +680,17 @@ WebsocketConnectionOptions ChargePoint::get_ws_connection_options(const int32_t
     }
 
     const auto network_connection_profile = network_connection_profile_opt.value();
-    auto ocpp_csms_url = network_connection_profile.ocppCsmsUrl.get();
 
-    if (ocpp_csms_url.compare(0, 5, "ws://") == 0) {
-        ocpp_csms_url.erase(0, 5);
-    } else if (ocpp_csms_url.compare(0, 6, "wss://") == 0) {
-        ocpp_csms_url.erase(0, 6);
-    }
+    auto security_profile = network_connection_profile.securityProfile;
+    auto uri = Uri::parse_and_validate(
+        network_connection_profile.ocppCsmsUrl.get(),
+        this->device_model->get_value<std::string>(ControllerComponentVariables::SecurityCtrlrIdentity),
```

Comment:
Identity is defined as not required in OCPP2.0.1 spec (so it should be requested from storage with get_optional_value). This seems to be a problem with the spec. Lets define Identity as requried in [config/](https://github.com/EVerest/libocpp/blob/main/config/v201/component_schemas/standardized/SecurityCtrlr.json#L174) and keep requesting storage with get_value

Reply:
Is there sth. more needed except adding `Identity` to `required:` in `SecurityCtrlr.json`?